parser/parser.y

```diff
@@ -2851,19 +2854,35 @@ FunctionCallConflict:
 		$$ = &ast.BinaryOperationExpr{Op: opcode.Mod, L: $3.(ast.ExprNode), R: $5.(ast.ExprNode)}
 	}
 
+DistinctKwd:
+	"DISTINCT"
+|	"DISTINCTROW"
+
 DistinctOpt:
+	"ALL"
 	{
 		$$ = false
 	}
-|	"ALL"
+|	DistinctKwd
+	{
+		$$ = true
+	}
+
```

Comment:
`"DISTINCT" "ALL"` moved to `BuggyXxxxx` since normally those keywords should not appear together.